src/hooks/Revolver.tsx

```diff
@@ -0,0 +1,33 @@
+import {KeyLayout, KeyRevolver} from "../types/tsTypes";
+import {useState} from "react";
+
+export default function useRevolver(initialRevolver: KeyRevolver) {
+    const [chamberInFocus, setChamberInFocus] = useState(0);
+    const [revolver, setRevolver] = useState(initialRevolver)
+
+    function deleteChamber(chamberToDelete: number) {
+        if (revolver[chamberToDelete]) revolver.splice(chamberToDelete, 1)
```

Comment:
I think it's not a safe method to manipulate the state like this. Usually you would want to do something like setRevolver(revolver.slice(chamberToDelete, revolver.length - 1)